### PR TITLE
chore: upgrade dependences to support react 18.X

### PR DESCRIPTION
**Pull request by rohit-till**
## Related Issues

Issue: #XXX

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,34 +39,47 @@
...
-    "react": ">=16.8.3",
-    "react-native": ">=0.59.0"
+    "react": "*",
+    "react-native": ">=0.65.0"
   },
   "devDependencies": {
-    "@babel/preset-env": "7.12.1",
-    "@babel/preset-react": "7.12.1",
-    "@babel/preset-typescript": "7.12.1",
+    "@babel/core": "7.13.15",
```
Comment (mdsiddiq-till) [inline package.json:49]: What is this change for? adding a lot of dependencies.

File: package.json [hunk 2]
```diff
@@ -39,34 +39,47 @@
...
   },
   "dependencies": {
-    "react-native-modal": "11.5.6"
+    "react-native-modal": "13.0.1"
   },
   "peerDependencies": {
-    "react": ">=16.8.3",
-    "react-native": ">=0.59.0"
+    "react": "*",
+    "react-native": ">=0.65.0"
```
Comment (mdsiddiq-till) [inline package.json:46]: So, this works for current react native version in pos and upcoming version change?

File: package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "modal",
     "useModal"
   ],
-  "version": "0.1.9",
+  "version": "0.1.10",
```
Comment (mdsiddiq-till) [inline package.json:11]: Let's go with 0.2.0 or 1.0.0?

File: package.json (hunk 1 above)
Comment (rohit-till) [reply package.json:49]: it's for unit testing, 

before adding:
https://github.com/oolio-group/rn-use-modal/actions/runs/7028800357/job/19125360359

File: package.json (hunk 2 above)
Comment (rohit-till) [reply package.json:46]: Yes, Tested on my local

File: package.json (hunk 3 above)
Comment (rohit-till) [reply package.json]: updated to 0.2.0

**Review by mdsiddiq-till — APPROVED** [review]

**Pull request merged**